### PR TITLE
[Dehoister] Add user blacklisting to various dehoister events and commands

### LINKED ISSUE #61: [Dehoister] User blacklisting
So the dehoister cog looks interesting, but there's just one problem:

Bots are picked up as hoisted users. Okay, maybe I should be more specific. Bots that have the name with the format of "PREFIX | BOTNAME" are picked up, and I don't want them to be. It would be nice if there was a way to either ignore bots or blacklist an entire role (or roles) so they're ignored.

### PR DESCRIPTION
**Pull request by Just-Jojo**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature
- [ ] Documentation

### Description of the changes
Adds an ignore list to have the dehoister ignore certain users. Doc strings might not be the best but it's me (:

If anything needs to be changed then lmk :D

Closes #61

### CODE REVIEW

File: dehoister/dehoister.py [hunk 1]
```diff
@@ -9,16 +9,19 @@
...
 
+from typing import Union
+
 log = logging.getLogger("red.kreusada.dehoister")
 
 IDENTIFIER = 435089473534
 
-HOIST = "!\"#$%&'()*+,-./:;<=>?@"
+HOIST = tuple("!\"#$%&'()*+,-./:;<=>?@")
+# Since every time this got used it was converted to a tuple just make a tuple here
```
Comment (Kreusada) [inline dehoister/dehoister.py:19]:
```suggestion
```
We can remove this comment here, thanks for letting me know about it though, makes sense.

File: dehoister/dehoister.py [hunk 2]
```diff
@@ -217,6 +214,7 @@ async def clean(self, ctx: commands.Context):
         """
         nickname = await self.config.guild(ctx.guild).nickname()
         hoisted_count = self.get_hoisted_count(ctx)
+        ignored_list = await self.config.guild(ctx.guild).ignored_users
```
Comment (Kreusada) [inline dehoister/dehoister.py:217]:
```suggestion
        ignored_list = await self.config.guild(ctx.guild).ignored_users()
```

File: dehoister/dehoister.py [hunk 3]
```diff
@@ -266,6 +264,53 @@ async def clean(self, ctx: commands.Context):
...
+        Add and remove certain users from being ignored by the auto-dehoister.
+        """
+
+    @ignore.command(name="add", require_var_positional=True)
+    async def ignore_add(self, ctx: commands.Context, *users: Union[discord.Member, int]):
+        """
+        Add users to the ignore list.
+
+        If on the ignore list, the dehoister will not dehoist them if they have a hoisted nickname
+        """
```
Comment (Kreusada) [inline dehoister/dehoister.py:279]:
```suggestion
        """Add users to the ignore list.

        If on the ignore list, the dehoister will not dehoist them if 
        they have a hoisted nickname/username.
        """
```

File: dehoister/dehoister.py [hunk 4]
```diff
@@ -266,6 +264,53 @@ async def clean(self, ctx: commands.Context):
...
+                    uid = uid.id
+                ignored_users.append(uid)
+        await ctx.tick()
+
+    @ignore.command(name="remove", aliases=["del"], require_var_positional=True)
+    async def ignore_remove(self, ctx: commands.Context, *users: int):
+        """
+        Remove users from the ignore list.
+        Once removed, they will be dehoisted by the dehoister if they have a hoisted nick
+        """ # This repetition is a bore >:|
```
Comment (Kreusada) [inline dehoister/dehoister.py:292]:
```suggestion
        """Remove users from the ignore list.
        
        Once removed, they will be dehoisted by the dehoister if 
        they have a hoisted nickname/username.
        """
```

File: dehoister/dehoister.py [hunk 5]
```diff
@@ -266,6 +264,53 @@ async def clean(self, ctx: commands.Context):
...
+            for uid in users:
+                try:
+                    ignored_users.remove(uid)
+                except ValueError:
+                    pass
+        await ctx.tick()
+
+    @ignore.command(name="list")
+    async def ignore_list(self, ctx: commands.Context):
+        """List the users ignored by the auto-dehoister"""
```
Comment (Kreusada) [inline dehoister/dehoister.py:303]:
```suggestion
        """List the users ignored by the auto-dehoister."""
```
:)

File: dehoister/dehoister.py [hunk 6]
```diff
@@ -266,6 +264,53 @@ async def clean(self, ctx: commands.Context):
     async def _set(self, ctx: commands.Context):
         """Settings for Dehoister."""
 
+    @_set.group()
+    async def ignore(self, ctx: commands.Context):
+        """
+        Add and remove certain users from being ignored by the auto-dehoister.
+        """
```
Comment (Kreusada) [inline dehoister/dehoister.py:271]:
```suggestion
        """Add and remove certain users from being ignored by the auto-dehoister."""
```
Just for styling here

**Review by Kreusada — CHANGES_REQUESTED** [review]
A few adjustments needed for this to be merged. Also, I'd like to see this ignore list respect the following commands:

- `[p]hoist clean`
- `[p]hoist scan`

Once that's done, should be all good to go. Thanks for your hard work <3

**Comment by Just-Jojo** [comment]
Changes have been pushed 👍

**Comment by Kreusada** [comment]
Looks good to me from a quick look, have these changes been tested?

**Comment by Just-Jojo** [comment]
They have indeed!

**Review by Kreusada — APPROVED** [review]
All set now, thanks for your patience. This looks great 👍

**Pull request merged**